wagtail/contrib/wagtailapi/serializers.py

```diff
@@ -93,80 +88,90 @@ def serialize_object_metadata(self, request, obj, show_details=False):
+        return super(BaseSerializer, self).to_representation(obj)
+
+    def to_representation(self, obj):
         data = [
             ('id', obj.id),
         ]
 
-        # Add meta
-        metadata = self.serialize_object_metadata(request, obj, show_details=show_details)
+        # Serialize meta
+        metadata = self.serialize_meta(obj)
         if metadata:
             data.append(('meta', metadata))
 
-        # Add extra data
-        data.extend(extra_data)
-
-        # Add other fields
-        api_fields = self.context['view'].get_api_fields(type(obj))
-        api_fields = list(OrderedDict.fromkeys(api_fields)) # Removes any duplicates in case the user put "title" in api_fields
-
-        if all_fields:
-            fields = api_fields
-        else:
-            unknown_fields = fields - set(api_fields)
-
-            if unknown_fields:
-                raise BadRequestError("unknown fields: %s" % ', '.join(sorted(unknown_fields)))
-
-            # Reorder fields so it matches the order of api_fields
-            fields = [field for field in api_fields if field in fields]
-
-        data.extend(get_api_data(obj, fields))
+        # Serialize fields
+        data.extend(self.serialize_fields(obj).items())
 
         return OrderedDict(data)
 
 
-class PageSerializer(WagtailSerializer):
-    def serialize_object_metadata(self, request, page, show_details=False):
-        data = super(PageSerializer, self).serialize_object_metadata(request, page, show_details=show_details)
+class PageSerializer(BaseSerializer):
+    def serialize_meta(self, page):
+        data = super(PageSerializer, self).serialize_meta(page)
 
         # Add type
         data['type'] = page.specific_class._meta.app_label + '.' + page.specific_class.__name__
 
         return data
 
-    def serialize_object(self, request, page, fields=frozenset(), extra_data=(), all_fields=False, show_details=False):
-        # Add parent
-        if show_details:
+    def serialize_fields(self, page):
+        data = list(super(PageSerializer, self).serialize_fields(page).items())
+
+        # Add parent field to the beginning of the fields section
+        if self.context.get('show_details', False):
             parent = page.get_parent()
 
-            site_pages = pages_for_site(request.site)
+            site_pages = pages_for_site(self.context['request'].site)
             if site_pages.filter(id=parent.id).exists():
                 parent_class = parent.specific_class
 
-                extra_data += (
+                data.insert(0,
                     ('parent', OrderedDict([
                         ('id', parent.id),
                         ('meta', OrderedDict([
                              ('type', parent_class._meta.app_label + '.' + parent_class.__name__),
                              ('detail_url', ObjectDetailURL(parent_class, parent.id)),
                         ])),
-                    ])),
+                    ]))
```

Comment:
Again, is there an alternative where 'parent' is a regular field, rather than overriding `serialize_fields`? (eg add it to the fields in `__init__` based on show_details)